radicle-cli/src/commands/delegate/add.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::Context as _;
+use radicle::{
+    prelude::{Did, Id},
+    storage::{WriteRepository as _, WriteStorage},
+    Profile,
+};
+use radicle_crypto::PublicKey;
+
+use crate::terminal as term;
+
+pub fn run<S>(profile: &Profile, storage: &S, id: Id, key: PublicKey) -> anyhow::Result<()>
+where
+    S: WriteStorage,
+{
+    let signer = term::signer(profile)?;
+    let me = signer.public_key();
+
+    let mut project = storage
+        .get(&profile.public_key, id)?
+        .context("No project with such ID exists")?;
+
+    let repo = storage.repository(id)?;
+
+    if !project.is_delegate(me) && key != *me {
+        return Err(anyhow::anyhow!("'{}' is not a current delegate of the project and does not have permission to add another person's key. However, they may add their own.", me));
+    }
```

Comment:
This should be changed now to only check if you're a delegate and just say you're not a delegate as the error.

Reply:
Also the `key != me` should be removed now. We do probably want to make sure that the key being added isn't already in the delegate set, ie. `!project.is_delegate(key)`